gabbi/httpclient.py

```diff
@@ -138,11 +139,20 @@ def _print_headers(self, headers, prefix=''):
 
     def _print_body(self, headers, content):
         """Output body if not binary."""
-        if self._show_body and utils.not_binary(
-                utils.extract_content_type(headers)[0]):
+        content_type = utils.extract_content_type(headers)[0]
+        if self._show_body and utils.not_binary(content_type):
+            content = utils.decode_response_content(headers, content)
+            # TODO(cdent): Using the JSONHandler here instead of
+            # just the json module to make it clear that eventually
+            # we could pretty print any printable output by using a
+            # handler's loads() and dumps(). Not doing that now
+            # because it would be pointless (no other interesting
+            # handlers) and this may approach may be entirely wrong.
```

Comment:
Sounds like you're doing a lot of interesting gabbi stuff. If you get a chance to share more about it, that would be great.

I've started an extracted HTML content handler over at https://github.com/cdent/gabbihtml based on stuff that @FND wrote a while ago. Having a diverse collection of such things that's relatively easy to find would be a nice to have, eventually.